lib/collection/src/shards/local_shard/shard_ops.rs

```diff
@@ -72,19 +72,25 @@ impl ShardOperation for LocalShard {
             // It is *critical* to hold `_wal_lock` while sending operation to the update handler!
             //
             // TODO: Refactor `lock_and_write`, so this is less terrible? :/
-            let (operation_id, _wal_lock) = match self.wal.lock_and_write(&mut operation).await {
-                Ok(id_and_lock) => id_and_lock,
+            let (operation_id, _wal_lock) = if let Some(ref wal) = self.wal {
+                match wal.lock_and_write(&mut operation).await {
+                    Ok(id_and_lock) => id_and_lock,
 
-                Err(shard::wal::WalError::ClockRejected) => {
-                    // Propagate clock rejection to operation sender
-                    return Ok(UpdateResult {
-                        operation_id: None,
-                        status: UpdateStatus::ClockRejected,
-                        clock_tag: operation.clock_tag,
-                    });
-                }
+                    Err(shard::wal::WalError::ClockRejected) => {
+                        // Propagate clock rejection to operation sender
+                        return Ok(UpdateResult {
+                            operation_id: None,
+                            status: UpdateStatus::ClockRejected,
+                            clock_tag: operation.clock_tag,
+                        });
+                    }
 
-                Err(err) => return Err(err.into()),
+                    Err(err) => return Err(err.into()),
+                }
+            } else {
+                return Err(CollectionError::service_error(
```

Comment:
maybe bad_request instead of service_error? Service error usually means that something unexpected happens on server-side.